backends/cloud/box/boxlistdirectorybyidrequest.cpp

```diff
@@ -0,0 +1,169 @@
+}
+
+void BoxListDirectoryByIdRequest::responseCallback(Networking::JsonResponse response) {
+	_workingRequest = nullptr;
+	if (_ignoreCallback) return;
+	if (response.request) _date = response.request->date();
+
+	Networking::ErrorResponse error(this);
+	Networking::CurlJsonRequest *rq = (Networking::CurlJsonRequest *)response.request;
+	if (rq && rq->getNetworkReadStream())
+		error.httpResponseCode = rq->getNetworkReadStream()->httpResponseCode();
+
+	Common::JSONValue *json = response.value;
+	if (json) {
+		Common::JSONObject responseObject = json->asObject();
+
+		//debug("%s", json->stringify(true).c_str());
+
+		//TODO: check that error is returned the right way
+		/*
+		if (responseObject.contains("error") || responseObject.contains("error_summary")) {
+		    warning("Box returned error: %s", responseObject.getVal("error_summary")->asString().c_str());
+		    error.failed = true;
+		    error.response = json->stringify();
+		    finishError(error);
+		    delete json;
+		    return;
+		}
+		*/
+
+		//TODO: check that ALL keys exist AND HAVE RIGHT TYPE to avoid segfaults
+
+		if (responseObject.contains("entries") && responseObject.getVal("entries")->isArray()) {
+			Common::JSONArray items = responseObject.getVal("entries")->asArray();
+			for (uint32 i = 0; i < items.size(); ++i) {
+				Common::JSONObject item = items[i]->asObject();
+				Common::String id = item.getVal("id")->asString();
+				Common::String name = item.getVal("name")->asString();
+				bool isDirectory = (item.getVal("type")->asString() == "folder");
+				uint32 size = 0, timestamp = 0;
+				if (item.contains("size")) {
+					if (item.getVal("size")->isString())
+						size = item.getVal("size")->asString().asUint64();
+					else if (item.getVal("size")->isIntegerNumber())
+						size = item.getVal("size")->asIntegerNumber();
+					else
+						warning("strange type for field 'size'");
+				}
+				if (item.contains("modified_at") && item.getVal("modified_at")->isString())
+					timestamp = ISO8601::convertToTimestamp(item.getVal("modified_at")->asString());
+
+				//as we list directory by id, we can't determine full path for the file, so we leave it empty
+				_files.push_back(StorageFile(id, "", name, size, timestamp, isDirectory));
+			}
+		}
+
+		uint32 received = 0;
+		uint32 totalCount = 0;
+		if (responseObject.contains("total_count") && responseObject.getVal("total_count")->isIntegerNumber())
+			totalCount = responseObject.getVal("total_count")->asIntegerNumber();
+		if (responseObject.contains("offset") && responseObject.getVal("offset")->isIntegerNumber())
+			received = responseObject.getVal("offset")->asIntegerNumber();
+		if (responseObject.contains("limit") && responseObject.getVal("limit")->isIntegerNumber())
+			received += responseObject.getVal("limit")->asIntegerNumber();
+		bool hasMore = (received < totalCount);
+
+		if (hasMore) makeRequest(received);
+		else finishListing(_files);
+	} else {
+		warning("null, not json");
```

Comment:
Please put prefix, like: "BoxListDirectoryByIdRequest::responseCallback(): null, not json". Will help greatly with the troubleshooting.